flickit-assessment-data/src/main/java/org/flickit/assessment/data/jpa/kit/kitversion/KitVersionJpaRepository.java

```diff
@@ -17,8 +17,11 @@ public interface KitVersionJpaRepository extends JpaRepository<KitVersionJpaEnti
     @Modifying
     @Query("""
             UPDATE KitVersionJpaEntity k
-            SET k.status = :status
+            SET k.status = :status,
+                k.statusVersion = :statusVersion
             WHERE k.id = :id
```

Comment:
**Suggestion:** Ensure that the update query in `updateStatus` method correctly handles the edge case where `statusVersion` might be set to a negative value unintentionally due to incorrect status ID or kitVersionId. [possible issue, importance: 7]
```suggestion
            UPDATE KitVersionJpaEntity k
                    SET k.status = :status,
                        k.statusVersion = CASE WHEN :statusVersion < 0 THEN 0 ELSE :statusVersion END
                    WHERE k.id = :id
```